src/Data/Bifunctor/TH.hs

```diff
@@ -1122,9 +1124,12 @@ functorLikeTraverse tvMap FT { ft_triv = caseTrivial,     ft_var = caseVar
       let (f, args) = unapplyTy t
       (_,   fc)  <- go co f
       (xrs, xcs) <- unzip <$> mapM (go co) args
-      let numLastArgs, numFirstArgs :: Int
-          numLastArgs  = min 2 $ length args
-          numFirstArgs = length args - numLastArgs
+      let numFirstArgs :: Int
+          mentionsMappedVar :: Type -> Bool
+          mentionsMappedVar = any (`Map.member` tvMap) . freeVariables
+          (firstArgs, lastArgs) = break mentionsMappedVar args
+          numFirstArgs = length firstArgs
+          numLastArgs = length lastArgs
```

Comment:
This is performing redundant work, as `go` is already computing whether each argument to the type constructor contains occurrences of the last two type parameters. A more efficient way to implement this would be something like this:

```hs
      let (xcFirsts, xcLasts) = break id xcs
          numLastArgs, numFirstArgs :: Int
          numFirstArgs = length xcFirsts
          numLastArgs = length xcLasts
```